### PR TITLE
Add subscriber autoreview tests

### PR DESCRIPTION
**Pull request by theofidry**
See the test in question for the new rules. No change on the code side, just making adding more error-friendly messages when the conventions are not respected

### CODE REVIEW

File: tests/phpunit/AutoReview/Event/SubscriberTest.php [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+use ReflectionClass;
+use ReflectionMethod;
+use ReflectionNamedType;
+use function Safe\array_flip;
+use function Safe\sprintf;
+
+final class SubscriberTest extends TestCase
+{
+    /**
+     * @dataProvider Infection\Tests\AutoReview\Event\SubscriberProvider::subscriberClassesProvider
```
Comment (localheinz) [inline tests/phpunit/AutoReview/Event/SubscriberTest.php]:
```suggestion
     * @dataProvider \Infection\Tests\AutoReview\Event\SubscriberProvider::subscriberClassesProvider
```

File: tests/phpunit/AutoReview/Event/SubscriberTest.php [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+            if (!$this->isSubscriptionMethod($method)) {
+                continue;
+            }
+
+            $this->assertIsSubscriptionMethod($subscriberClass, $method);
+        }
+    }
+
+    /**
+     * @dataProvider Infection\Tests\AutoReview\Event\SubscriberProvider::subscriberClassesProvider
```
Comment (localheinz) [inline tests/phpunit/AutoReview/Event/SubscriberTest.php]:
```suggestion
     * @dataProvider \Infection\Tests\AutoReview\Event\SubscriberProvider::subscriberClassesProvider
```

File: tests/phpunit/AutoReview/Event/SubscriberTest.php [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+        /** @var EventSubscriber $subscriber */
+        $subscriber = (new ReflectionClass($subscriberClass))->newInstanceWithoutConstructor();
+
+        $subscriber->getSubscribedEvents();
+
+        $this->addToAssertionCount(1);
+    }
+
+    /**
+     * @dataProvider Infection\Tests\AutoReview\Event\SubscriberProvider::subscriberSubscriptionMethodsProvider
```
Comment (localheinz) [inline tests/phpunit/AutoReview/Event/SubscriberTest.php]:
```suggestion
     * @dataProvider \Infection\Tests\AutoReview\Event\SubscriberProvider::subscriberSubscriptionMethodsProvider
```

File: tests/phpunit/AutoReview/Event/SubscriberTest.php [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+            && $method->getName() !== 'getSubscribedEvents'
+        ;
+    }
+
+    /**
+     * @param class-string $subscriberClass
+     */
+    private function assertIsSubscriptionMethod(string $subscriberClass, ReflectionMethod $method): void
+    {
+        $this->assertSame(
```
Comment (localheinz) [inline tests/phpunit/AutoReview/Event/SubscriberTest.php]: Doesn't matter much, but perhaps using `assertCount()` here makes a bit more sense?

```diff
diff --git a/tests/phpunit/AutoReview/Event/SubscriberTest.php b/tests/phpunit/AutoReview/Event/SubscriberTest.php
index e38d284e..db789702 100644
--- a/tests/phpunit/AutoReview/Event/SubscriberTest.php
+++ b/tests/phpunit/AutoReview/Event/SubscriberTest.php
@@ -36,6 +36,7 @@ declare(strict_types=1);
 namespace Infection\Tests\AutoReview\Event;

 use function array_keys;
+use function count;
 use function implode;
 use Infection\Event\Subscriber\EventSubscriber;
 use const PHP_EOL;
@@ -143,15 +144,15 @@ final class SubscriberTest extends TestCase
      */
     private function assertIsSubscriptionMethod(string $subscriberClass, ReflectionMethod $method): void
     {
-        $this->assertSame(
+        $this->assertCount(
             1,
-            $method->getNumberOfParameters(),
+            $method->getParameters(),
             sprintf(
                 'Expected a subscription method "%s::%s" to have exactly one parameter: the'
                 . ' event. Got "%d"',
                 $subscriberClass,
                 $method->getName(),
-                $method->getNumberOfParameters()
+                count($method->getNumberOfParameters())
             )
         );

```

Comment (theofidry) [reply tests/phpunit/AutoReview/Event/SubscriberTest.php]: Oups thought you were done with the review

I think it's ok: what matters here is the error message. I don't have a strong preference but one is slightly more verbose than the other so might go for the shorter version

**Pull request merged**